api/apps/api/src/modules/projects/edit-guard/marxan-edit-guard.service.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { Test } from '@nestjs/testing';
+import { EditGuard } from '@marxan-api/modules/projects/edit-guard/edit-guard.service';
+import { ProjectChecker } from '@marxan-api/modules/projects/project-checker/project-checker.service';
+import { ProjectCheckerFake } from '../../../../test/utils/project-checker.service-fake';
+import { v4 } from 'uuid';
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { MarxanEditGuard } from '@marxan-api/modules/projects/edit-guard/marxan-edit-guard.service';
+
+describe('MarxanEditGuard', () => {
+  let fixtures: FixtureType<typeof getFixtures>;
+
+  beforeEach(async () => {
+    fixtures = await getFixtures();
+  });
+
+  it(`throws an exception if the given project has an ongoing export`, async () => {
+    const projectId = fixtures.GivenProjectWasCreated();
+
+    await fixtures.WhenExportIsRequested(projectId);
+
+    await fixtures.ThenAnExceptionIsThrown(projectId);
```

Comment:
usual suggestion for clarity on what we are trying to assert...

```suggestion
    await fixtures.ThenAnExceptionIsThrownIfCheckingWhetherProjectCanBeEdited(projectId);
```
In practice I'd argue this should be broken down again in another action (e.g. `WhenCheckingWhetherTheProjectCanBeEdited()`) + assertion (`ThenAnExceptionIsThrown()`), but I'd leave the decision up to you.

Reply:
I had the same doubt, I think is more correct splitting it in two steps 👌🏽